beacon-chain/p2p/service.go

```diff
@@ -396,9 +397,14 @@ func (s *Service) AddPingMethod(reqFunc func(ctx context.Context, id peer.ID) er
 func (s *Service) pingPeers() {
 	s.pingMethodLock.RLock()
 	defer s.pingMethodLock.RUnlock()
+
+	localENR := s.dv5Listener.Self()
+	log.WithField("ENR", localENR).Info("New node record")
```

Comment:
Why not?

We log the very first ENR here:
```
INFO p2p: Started discovery v5
ENR=enr:-Me4QOjJD9yIdENgUqCEWvF8-i4ptgYCcE2_-585RTOkZmTbH09IuXNW5Y3hX_WG_-DEncEytbRry_tLe31BECrKPRGGAZG3Q7i0h2F0dG5ldHOIAAAAAAAAAACDY3NjBIRldGgykDdW2IZgAAA4AOH1BQAAAACCaWSCdjSCaXCErBAAEolzZWNwMjU2azGhA4Vo-9vK3iXroeg9TmKANgrXuntjWj6atypZL7bbpjBviHN5bmNuZXRzAIN0Y3CCMsiDdW
RwgjLI
```

But it turns out this ENR is the real one for a very small amount of time.
This ENR is modified shortly after.

Logging the initial ENR without logging modified ones is quite misleading since the user will think that it is it actual ENR, while it is not any more the case.

My opinion is:
- We should log all ENR changes, or
- We should NOT log any ENR at all.

Reply:
If you do want to log enr changes, `pingPeers` is the wrong place to do it.  It is more accurate to do it in `RefreshPersistentSubnets` as that is where the enr is actually modified.

Reply:
I agree.
However, currently `pingPeers` in only called by `RefreshPersistentSubnets`.
And the way `RefreshPersistentSubnets` is coded, we should add this log at three different locations, instead of only one.

I can of course do it.

Reply:
I can also rename `pingPeers` into `pingPeersAndLogEnr`.

Reply:
renaming would be better then

Reply:
Fixed in bd82d71d6bec2cd8b86da56086f5aaa656814325.